packages/teleport/src/Databases/AddDatabase/AddDatabase.test.tsx

```diff
@@ -46,28 +55,61 @@ describe('correct database add command generated with given input', () => {
 });
 
 test('correct tsh login command generated with local authType', () => {
-  render(<AddDialog {...props} />);
+  render(<AddDatabase {...props} />);
   const output = 'tsh login --proxy=localhost:443 --auth=local --user=yassine';
 
   expect(screen.queryByText(output)).not.toBeNull();
 });
 
 test('correct tsh login command generated with sso authType', () => {
-  render(<AddDialog {...props} authType="sso" />);
+  render(<AddDatabase {...props} authType="sso" />);
   const output = 'tsh login --proxy=localhost:443';
 
   expect(screen.queryByText(output)).not.toBeNull();
 });
 
+test('show correct steps when token attempt succeeded', () => {
+  render(
+    <AddDatabase {...props} authType="sso" attempt={{ status: 'success' }} />
+  );
+
+  expect(screen.queryByText('Step 1')).not.toBeNull();
+  expect(screen.queryByText('Step 2')).not.toBeNull();
+  expect(screen.queryByText('Step 3')).not.toBeNull();
+  expect(screen.queryByText('Step 4')).toBeNull();
+  expect(screen.queryByText('Step 5')).toBeNull();
+});
+
+test('show correct steps when token attempt failed', () => {
```

Comment:
this too is also redundant, the only thing i see different is that this test tests for `authType="sso"` render, i would add a storybook and snapshot that render to test this (much simpler, and more visual)

Reply:
Thanks, added a new entry and snapshot test in https://github.com/gravitational/webapps/pull/653/commits/3d4500c7c514c36f95505d23216b97bf541bebcf